modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -1143,6 +1144,10 @@ private void _validateName(long objectDefinitionId1, String name)
 		int count = objectRelationshipPersistence.countByODI1_N(
 			objectDefinitionId1, name);
 
+		count +=
+			_objectFieldPersistence.countByODI_N(objectDefinitionId1, name) +
+				_objectFieldPersistence.countByODI_N(objectDefinitionId2, name);
+
```

Comment:
I believe is not necessary count by `objectDefinitionId1`. Because the field is created only in children (`objectDefinitionId2`).

Also, we need add this validation after first if. Because we need show that the name already exist in other field of `objectDefinition2`. The current message in to duplicate name of relationship.

Reply:
Hey @murilostodolni , I saw that there is `ObjectRelationshipNameException` which is probably more adequate for this scenario. 
I also created a static [MustNotBeEqualToObjectRelationshipName](https://github.com/liferay-objects/liferay-portal/pull/2619/files#:~:text=public%20static%20class-,MustNotBeEqualToObjectRelationshipName,-extends%20ObjectFieldNameException%20%7B) exception for the validation in `ObjectFieldLocalService`. 

You think that makes sense?